core-bundle/src/Doctrine/Backup/BackupManager.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Doctrine\Backup;
+
+use Contao\CoreBundle\Doctrine\Backup\Config\AbstractConfig;
+use Contao\CoreBundle\Doctrine\Backup\Config\CreateConfig;
+use Contao\CoreBundle\Doctrine\Backup\Config\RestoreConfig;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Schema\Table;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Output\StreamOutput;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Webfactory\Slimdump\Config\Config;
+use Webfactory\Slimdump\Config\ConfigBuilder;
+use Webfactory\Slimdump\DumpTask;
+
+/**
+ * @internal
+ * @final
+ */
+class BackupManager
+{
+    private const DUMP_HEADER = '-- Generated by the Contao Open Source CMS Backup Manager (version: %s).';
+    private const DUMP_VERSION = 'v1';
+
+    private Connection $connection;
+    private string $backupDir;
+    private array $tablesToIgnore;
+    private int $keepMax;
+
+    public function __construct(Connection $connection, string $backupDir, array $tablesToIgnore, int $keepMax)
+    {
+        $this->connection = $connection;
+        $this->backupDir = $backupDir;
+        $this->tablesToIgnore = $tablesToIgnore;
+        $this->keepMax = $keepMax;
+    }
+
+    public function createCreateConfig(): CreateConfig
+    {
+        $config = new CreateConfig(Backup::createNewAtPath($this->backupDir));
+
+        return $config->withTablesToIgnore($this->tablesToIgnore);
+    }
+
+    /**
+     * @throws BackupManagerException
+     */
+    public function createRestoreConfig(): RestoreConfig
+    {
+        $latestBackup = $this->getLatestBackup();
+
+        if (null === $latestBackup) {
+            throw new BackupManagerException('No backups found.');
+        }
+
+        $config = new RestoreConfig($latestBackup);
+
+        return $config->withTablesToIgnore($this->tablesToIgnore);
+    }
+
+    public function getLatestBackup(): ?Backup
+    {
+        return $this->listBackups()[0] ?? null;
+    }
+
+    /**
+     * Lists all backups (latest one first).
+     *
+     * @return array<Backup>
+     */
+    public function listBackups(): array
+    {
+        (new Filesystem())->mkdir($this->backupDir); // Ensures, the directory exists
+
+        $backups = [];
+        $files = Finder::create()
+            ->files() // TODO: depth == 0?
+            ->in($this->backupDir)
+            ->ignoreDotFiles(true)
+            ->ignoreVCS(true)
+        ;
+
+        foreach ($files as $file) {
+            $backups[] = new Backup($file->getPathname());
+        }
+
+        uasort($backups, static fn (Backup $a, Backup $b) => $b->getCreatedAt() <=> $a->getCreatedAt());
```

Comment:
`Finder` allows for a sorting callback, maybe we should use that?

Reply:
No we cannot because we don't sort in the finder.